projects/plugins/jetpack/modules/masterbar/profile-edit/class-wpcom-user-profile-fields-revert.php

```diff
@@ -0,0 +1,144 @@
+<?php
+/**
+ * Manage User profile fields.
+ *
+ * @package automattic/jetpack
+ */
+
+namespace Automattic\Jetpack\Dashboard_Customizations;
+
+use Automattic\Jetpack\Connection\Manager as Connection_Manager;
+
+/**
+ * Responsible with preventing the back-end default implementation to save the fields that are managed on WP.com profiles.
+ *
+ * Class Profile_Edit_Filter_Fields
+ */
+class WPCOM_User_Profile_Fields_Revert {
+
+	/**
+	 * Jetpack connection manager object.
+	 *
+	 * @var Connection_Manager
+	 */
+	private $connection_manager;
+
+	/**
+	 * Profile_Edit_Filter_Fields constructor.
+	 *
+	 * @param string             $current_page The current page slug.
+	 * @param Connection_Manager $connection_manager The connection manager.
+	 */
+	public function __construct( $current_page, Connection_Manager $connection_manager ) {
+		// We'll apply this filter only on profile edit page to be sure we limit the impact of the change.
+		if ( 'user-edit.php' !== $current_page && 'profile.php' !== $current_page ) {
+			return;
+		}
```

Comment:
Nitpick: Now that we're hooking into specific actions, I'd suggest to remove this check. That way it'd be explicit that anytime that `WPCOM_User_Profile_Fields_Revert` is loaded it'd perform this logic.

Reply:
Yup, good point! :)